File: CS/Helpers/CallbackComboBoxHelper.cs
using DevExpress.Web;
using DevExpress.Web.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace E4425.Helpers {
    public static class CallbackComboBoxHelper {
        private static ComboBoxSettings callbackComboBox;
        public static ComboBoxSettings CreateCommonComboBoxSettings() {
            callbackComboBox = new ComboBoxSettings();
            callbackComboBox.Properties.TextField = "CountryName";
            callbackComboBox.Properties.ValueField = "CountryID";
            callbackComboBox.Properties.ValueType = typeof(int);
            callbackComboBox.Properties.DropDownStyle = DropDownStyle.DropDown;
            callbackComboBox.Properties.CallbackPageSize = 20;
            return callbackComboBox;
        }
        public static ComboBoxSettings CreateComboBoxSettings(bool isInEditForm) {
            callbackComboBox = CreateCommonComboBoxSettings();
            if (isInEditForm) {
                callbackComboBox.Name = "CountryID";
                callbackComboBox.CallbackRouteValues = new { Controller = "Home", Action = "CountryComboBoxPartial" };
            } else {
                callbackComboBox.Name = "FilterCountryID";
                callbackComboBox.CallbackRouteValues = new { Controller = "Home", Action = "FilterComboBox" };
                callbackComboBox.Properties.ClientSideEvents.SelectedIndexChanged = "OnSelectedIndexChanged";
            }
            return callbackComboBox;
        }

    }

}

File: VB/Helpers/CallbackComboBoxHelper.vb
Imports DevExpress.Web
Imports DevExpress.Web.Mvc
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Namespace E4425.Helpers
	Public NotInheritable Class CallbackComboBoxHelper

		Private Sub New()
		End Sub
		Private Shared callbackComboBox As ComboBoxSettings

		Public Shared Function CreateComboBoxColumnProperties() As MVCxColumnComboBoxProperties
			Dim p As New MVCxColumnComboBoxProperties()
				p.CallbackRouteValues = New With {Key .Controller = "Home", Key .Action = "CountryComboBox"}
				p.ValueField = "CountryID"
				p.TextField = "CountryName"
				p.ValueType = GetType(Integer)
				p.CallbackPageSize = 20
				p.DropDownStyle = DropDownStyle.DropDown
				p.BindList(E4425.Models.DataProvider.GetCountries())
			Return p
		End Function

	End Class
End Namespace
